core/src/engine/state/cached.rs

```diff
@@ -417,7 +415,7 @@ impl CachedAccount {
     }
 
     #[inline]
-    pub fn clear_expired_nonce(&mut self, n: U256) -> bool {
-        self.nonces.clear_expired(n)
+    pub fn cleanup_nonce_by_prefix(&mut self, nonce: Nonce) -> bool {
```

Comment:
Let's be explicit about what prefix we're cleaning up by:
```suggestion
    pub fn cleanup_nonce_by_prefix(&mut self, prefix: [u8; 31]) -> bool {
```
So that the caller cannot make wrong assumptions on the length of the prefix that will be used